### ISSUE TITLE
Are model weights available?

### ISSUE THREAD

**Issue by han-yeol** [estenhl/pyment-public#9]
Hi

I have used your code for inference(test) using random sampled IXI dataset 
However, prediction is very confused.

,age,prediction
IXI607_reo_regi_crop-Guys-1097-T1_bet,1,3.0171993
IXI616_reo_regi_crop-Guys-1092-T1_bet,1,3.017546
IXI617_reo_regi_crop-Guys-1090-T1_bet,1,3.01455
IXI618_reo_regi_crop-Guys-1091-T1_bet,1,3.0149872
IXI619_reo_regi_crop-Guys-1099-T1_bet,1,3.0176623
IXI622_reo_regi_crop-Guys-1102-T1_bet,1,3.0178277
IXI623_reo_regi_crop-Guys-1076-T1_bet,1,3.0185328
IXI625_reo_regi_crop-Guys-1098-T1_bet,1,3.0163016
IXI626_reo_regi_crop-Guys-1094-T1_bet,1,3.0155559
IXI627_reo_regi_crop-Guys-1103-T1_bet,1,3.0175211
IXI629_reo_regi_crop-Guys-1095-T1_bet,1,3.0163457
IXI630_reo_regi_crop-Guys-1108-T1_bet,1,3.0049784
IXI639_reo_regi_crop-Guys-1088-T1_bet,1,3.0180109
IXI640_reo_regi_crop-Guys-1106-T1_bet,1,3.0169275
IXI641_reo_regi_crop-Guys-1105-T1_bet,1,3.0183208
IXI642_reo_regi_crop-Guys-1104-T1_bet,1,3.017383
IXI644_reo_regi_crop-Guys-1121-T1_bet,1,3.0184426
IXI648_reo_regi_crop-Guys-1107-T1_bet,1,3.0121117
IXI652_reo_regi_crop-Guys-1116-T1_bet,1,3.018441
IXI653_reo_regi_crop-Guys-1122-T1_bet,1,3.0163202
IXI662_reo_regi_crop-Guys-1120-T1_bet,1,3.0182319

Label is all allocated by 1 (bacause it is test)
preprocessing(bet,reorient,register,crop,normalize) 
is model weight available now? I think the model is random weight.

**Comment by estenhl** [comment]
Hi @han-yeol, thanks for getting in touch!

It seems to be working correctly here for me. Using the following script:

```
import os
import nibabel as nib
import numpy as np

from pyment.models import RegressionSFCN


model = RegressionSFCN(weights='brain-age', include_top=True)


IXI_PATH = os.path.join(os.path.expanduser('~'), 'data', 'IXI', 'cropped', 'images')
filenames = ['IXI324-Guys-0922-T1.nii.gz', 'IXI482-HH-2178-T1.nii.gz']

for filename in filenames:
	image = nib.load(os.path.join(IXI_PATH, filename))
	image = image.get_fdata()
	image = image / 255.

	prediction = model.predict(np.expand_dims(image, 0))[0,0]
	print(f'{filename}: {prediction}')
```
yields the following output: 
```
1/1 [==============================] - 2s 2s/step
IXI324-Guys-0922-T1.nii.gz: 57.33319854736328
1/1 [==============================] - 0s 36ms/step
IXI482-HH-2178-T1.nii.gz: 69.96650695800781
```

Can you either cross-check whether you are doing anything differently in your scripts, and if so it starts working again if you replicate what I did? Alternatively send me the code you are using to generate predictions? It is worth noting that I used FreeSurfer to skullstrip, not bet

**Comment by han-yeol** [comment]
Thank you for your advice.
Your code seems to work correctly

first of all, my preprocessing process is just simple (I try to use your preprocessing docker, but it is fail. sorry)
1. HD-BET to skull-stripping (brain extraction) (2019, human-brain-mapping) (https://github.com/MIC-DKFZ/HD-BET)
2. fsl reorient
3. fsl linear registraion (dof 6)
4. crop the background (fslroi)
5. Intensity normalization (min-max scaling in your code)

,age,prediction
IXI607_reo_regi_crop-Guys-1097-T1_bet,1,18.371407
IXI616_reo_regi_crop-Guys-1092-T1_bet,1,15.590334
IXI617_reo_regi_crop-Guys-1090-T1_bet,1,14.33045
IXI618_reo_regi_crop-Guys-1091-T1_bet,1,15.085575
IXI619_reo_regi_crop-Guys-1099-T1_bet,1,15.371148
IXI622_reo_regi_crop-Guys-1102-T1_bet,1,16.567806
IXI623_reo_regi_crop-Guys-1076-T1_bet,1,17.003582
IXI625_reo_regi_crop-Guys-1098-T1_bet,1,15.589634
IXI626_reo_regi_crop-Guys-1094-T1_bet,1,13.419614
IXI627_reo_regi_crop-Guys-1103-T1_bet,1,17.773106
IXI629_reo_regi_crop-Guys-1095-T1_bet,1,13.556765
IXI630_reo_regi_crop-Guys-1108-T1_bet,1,50.204346
IXI639_reo_regi_crop-Guys-1088-T1_bet,1,16.98963
IXI640_reo_regi_crop-Guys-1106-T1_bet,1,14.146122
IXI641_reo_regi_crop-Guys-1105-T1_bet,1,16.883896
IXI642_reo_regi_crop-Guys-1104-T1_bet,1,15.041327
IXI644_reo_regi_crop-Guys-1121-T1_bet,1,18.961208
IXI648_reo_regi_crop-Guys-1107-T1_bet,1,14.574279
IXI652_reo_regi_crop-Guys-1116-T1_bet,1,18.207335
IXI653_reo_regi_crop-Guys-1122-T1_bet,1,17.387487
IXI662_reo_regi_crop-Guys-1120-T1_bet,1,16.381767

**Comment by han-yeol** [comment]
Plus Question about your paper 

In your paper Fig 5
You said that "The first used age and sex as predictor

From what I understand, You train SFCN model to predict age or sex without any pretraining weight.

Is this correct ? 

thank you for your code and paper

**Comment by estenhl** [comment]
That portion of the paper refers to the models that were used for case-control classifications, e.g. just simple linear models. The variables that are mentioned there are what predictors were used in the linear model. Hope this clarifies :) Regarding your earlier comments, is the code working for you now?

**Comment by han-yeol** [comment]
@estenhl yes, but we need to verify output and real age (in xls file)
Thank you very much!

**Comment by estenhl** [comment]
Good, then I will close the issue :) Reach out again if you need any more help
